### PR TITLE
Feature/login

### PR DESCRIPTION
**Pull request by Satvik26**
Create the UI for the login page.

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -5,7 +5,21 @@ export default {
...
+      'custom-grey' : '#D9D9D9',
+      'custom-green' : '#10CA00',
+      'custom-light-green' : '#E1FCEF',
+      'custom-yellow' : '#FFD700',
+      'custom-light-yellow' :'FEF7D1',
+      'custom-red': '#FF0000',
+      'custom-light-red' : '#FFD0D0',
+      'custom-black' : '#000000'
+
+    },},
```
Comment (DoStini) [inline tailwind.config.js:22]:
```suggestion
 extend: {
        colors: {
            'custom-light-blue': '#DEF2FF',
            'custom-blue' : '#0099FF',
            'custom-dard-blue' : '#003EAE',
            'custom-brown' : '#673E00',
            'custom-grey' : '#D9D9D9',
            'custom-green' : '#10CA00',
            'custom-light-green' : '#E1FCEF',
            'custom-yellow' : '#FFD700',
            'custom-light-yellow' :'FEF7D1',
            'custom-red': '#FF0000',
            'custom-light-red' : '#FFD0D0',
            'custom-black' : '#000000'
        },
      }
    },
```

**Review by DoStini — APPROVED** [review]

File: src/components/Login.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                    <h2 className='text-2xl font-bold text-center mb-8'>Login to Admin Panel</h2>
+                    <div>
+                      <label  className="block mb-2 text-sm font-medium text-gray-900 dark:text-gray">Email Address</label>
+                      <input type="email" name="email" id="email" className="mb-4 shadow-lg shadow-gray-300 border border-blue-500 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5" placeholder="name@company.com"  />
+                  </div>
+                  <div>
+                      <label  className="block mb-2 text-sm font-medium text-gray-900 dark:text-gray">Password</label>
+                      <input type="password" name="password" id="password" placeholder="••••••••" className="mb-3 shadow-lg shadow-gray-300 border border-blue-500 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5"  />
+                  </div>
+                    <button className='w-full py-2 my-4 bg-blue-600 hover:bg-blue-600 rounded-lg'>Login</button>
```
Comment (DoStini) [inline src/components/Login.tsx:24]: replace all blues with our custom ones

File: src/components/Login.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            <div className='p-10 pr-20 flex flex-col justify-around bg-white rounded-tr-2xl rounded-br-2xl'>
+                <form>
+                    <h2 className='text-2xl font-bold text-center mb-8'>Login to Admin Panel</h2>
+                    <div>
+                      <label  className="block mb-2 text-sm font-medium text-gray-900 dark:text-gray">Email Address</label>
+                      <input type="email" name="email" id="email" className="mb-4 shadow-lg shadow-gray-300 border border-blue-500 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5" placeholder="name@company.com"  />
+                  </div>
+                  <div>
+                      <label  className="block mb-2 text-sm font-medium text-gray-900 dark:text-gray">Password</label>
+                      <input type="password" name="password" id="password" placeholder="••••••••" className="mb-3 shadow-lg shadow-gray-300 border border-blue-500 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5"  />
```
Comment (DoStini) [inline src/components/Login.tsx:22]: you can use for example text-custom-blue and border-custom-blue for example

**Review by DoStini — CHANGES_REQUESTED** [review]

**Comment by Satvik26** [comment]
I have used the custom colors and added the authentication functionality. I was not able to modularise a way since I needed the variables for the post request. Do let me know your thoughts.

File: src/services/userAuth.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+export default api;
+
+api.interceptors.response.use(
+  (response) => response,
+  async (error) => {
+      const originalRequest = error.config;
+
+      // If the error status is 403, it means the token has expired and we need to refresh it
+      if (error.response.status === 403 && error.response.status === 422) {
```
Comment (DoStini) [inline src/services/userAuth.ts:30]: How is this ever gonna be true

Comment (Satvik26) [reply src/services/userAuth.ts:30]: This logic is for the refresh token authentication. It should run after the access token has expired. The refresh token logic is not working. As discussed in the call we can implement the refresh token in later sprint.

File: src/services/userAuth.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+              const refreshToken = Cookies.get('refreshToken');
+              console.log(refreshToken)
+              const response = await api.post('/refresh', {refresh_token:refreshToken } ,
+                  {
+                      headers: { 'Content-Type': 'application/json' },
+
+                  });
+              console.log('Refresh Token Response:', response);
+
+              const { new_accessToken, new_refreshToken } = response.data;
```
Comment (DoStini) [inline src/services/userAuth.ts:41]: this has to be access_token, refresh_token (response given by the server)

**Review by DoStini — APPROVED** [review]

**Pull request merged**